test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -331,3 +331,30 @@
    (with-puppet-conf puppet-conf-file conf-dir))
   ([puppet-conf-file dest-dir]
    (with-puppet-conf-fixture {"puppet.conf" puppet-conf-file} dest-dir)))
+
+(defn copy-config-files
+  [dirs-to-copy]
+  (doseq [[src target] dirs-to-copy]
+    (fs/mkdirs target)
+    (let [[_ dirs files] (first (fs/iterate-dir src))]
+      (doseq [d dirs] (fs/copy-dir (fs/file src d) target))
+      (doseq [f files] (fs/copy (fs/file src f) (fs/file target f))))))
+
+(defn remove-config-files
+  [dirs-to-copy]
+  (doseq [[src target] dirs-to-copy]
+    (let [[_ dirs files] (first (fs/iterate-dir src))]
+      (doseq [d dirs] (fs/delete-dir (fs/file target d)))
+      (doseq [f files] (fs/delete (fs/file target f))))))
+
+(defmacro with-puppet-config-files
```

Comment:
So I guess is the difference between this and the `with-puppet-conf-files` macro is that this works for not just the confdir but also the codedir? Would be nice at some point to clean these helper functions/macros up because it feels like we have a lot of functions doing similar things. But probably outside the scope of this PR?